include/eve/module/complex/regular/detail/arithmetic.hpp

```diff
@@ -0,0 +1,47 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/module/core.hpp>
+#include <eve/module/math.hpp>
+#include <eve/module/complex/regular/traits.hpp>
+#include <eve/module/complex/regular/conj.hpp>
+#include <eve/module/complex/regular/proj.hpp>
+
+namespace eve::detail
+{
+  EVE_FORCEINLINE auto complex_unary_dispatch(eve::tag::arg_, auto const& z) noexcept
```

Comment:
Why unary/binary? I mean, we have perfect forwarding.

Reply:
Do you have tests for functions that should work as default?
Like `eve::zip`?

Reply:
1./ I stumbled upon an ambiguous case for functions that may have 1 or 2 arguments AND supports decorator. SPlitting them this way was the easiest way I found to prevent those ambiguities. I can have another look if needed.

2./ Good point, will add one. I tested it locally but didn't write one.

Reply:
1. Ok - makes sense.